misaligned.cpp

```diff
@@ -1,21 +1,63 @@
 #include <iostream>
 #include <assert.h>
+#include <map>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+
+int calculatePairNo(int majorIndex, int minorIndex)
+{
+    return ((majorIndex * 5 + minorIndex) + 1); 
+}
+
+std::string createColorPair(int majorIndex, int minorIndex)
+{
+    std::string majorColorName(majorColor[majorIndex]);
+    std::string minorColorName(minorColor[minorIndex]);
+    return majorColorName.append(" | ").append(minorColorName);
+}
+
+int getPairNumber(int majorIndex, int minorIndex, std::map<int, std::string>& ColorPairs)
+{
+    int pair_no = calculatePairNo(majorIndex, minorIndex);
+    std::string colorPair = createColorPair(majorIndex, minorIndex);
+    ColorPairs[pair_no] = colorPair;
+    return pair_no;
+}
+
+void printColorCode(int majorIndex, int minorIndex, std::map<int, std::string>& ColorPairs)
+{
+    int pair_no = getPairNumber(majorIndex, minorIndex, ColorPairs);
+    std::cout << pair_no << " | " << ColorPairs[pair_no] << "\n";

```

Comment:
the alignment of printed line varies by one space for single digit and double digit pair number, 
ex -
1 | ...
10 | ...

Reply:
Updated the file. The alignment looks fine for all the digits now.